exonum/src/runtime/dispatcher/types.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Internal dispatcher data types
```

Comment:
These data types are not internal though (`ArtifactStatus` and `InstanceStatus` are exposed via `DispatcherSchema`). I'd argue that these types *should* be public and could be moved together with other runtime types.